### ISSUE TITLE
bug: default geometric smearing for generic detector is broken

### ISSUE THREAD

**Issue by gagnonlg** [acts-project/acts#2020]
Trying to configure the digitization of the generic detector with the default config (default-geometric-config-generic.json) currently leads to the following error:

```
Traceback (most recent call last):
  File "/home/glg/ACTS/worktrees/workspaces/pos-calibrate/pos-calibrate.py", line 103, in <module>
    main()
  File "/home/glg/ACTS/worktrees/workspaces/pos-calibrate/pos-calibrate.py", line 100, in main
    generate_trackstates(nevents=5000, rng=rng,)
  File "/home/glg/ACTS/worktrees/workspaces/pos-calibrate/pos-calibrate.py", line 52, in generate_trackstates
    acts.examples.simulation.addDigitization(
  File "/home/glg/ACTS/worktrees/pos-calibrate/build/python/acts/examples/simulation.py", line 718, in addDigitization
    acts.examples.readDigiConfigFromJson(
RuntimeError: [json.exception.type_error.305] cannot use operator[] with a string argument with binary
```
I traced it back to #1834 . This PR introduced a configurable charge smearer, and reads it unconditionally from the config. However the default configuration was not remade, leading to this error.

One possible fix is the following patch:
```
modified   Examples/Io/Json/src/JsonDigitizationConfig.cpp
@@ -138,7 +138,9 @@ void ActsExamples::from_json(const nlohmann::json& j,
   gdc.thickness = j["thickness"];
   gdc.threshold = j["threshold"];
   gdc.digital = j["digital"];
-  from_json(j["charge-smearing"], gdc.chargeSmearer);
+  if (j.contains("charge-smearing")) {
+      from_json(j["charge-smearing"], gdc.chargeSmearer);
+  }
 }
 
 void ActsExamples::to_json(nlohmann::json& j,

```

Alternatively the default config file could simply be remade, but I don't know how to do that.

Pinging @benjaminhuth @asalzburger @paulgessinger , thoughts? If the above fix is fine, I can submit a PR but otherwise one of you needs to remake the default config (or teach me how to do it).

**Comment by gagnonlg** [comment]
@benjaminhuth in a sense the default config is there to "showcase" what our software can do so I think it'd make sense to go with the first alternative?

**Comment by benjaminhuth** [comment]
Sounds good to me! Maybe also worth adding a CI check to prevent future breaking changes.

**Comment by benjaminhuth** [comment]
I can do it but won't have time today, probably tomorrow or early next week.

**Comment by gagnonlg** [comment]
Great, thanks!

**Comment by gagnonlg** [comment]
Still relevant, will be fixed by  #2081
